_source/_posts/2023-01-24-jakartaee-auth0.md

```diff
@@ -0,0 +1,545 @@
+---
+layout: blog_post
+title: "Open ID authentication with Jakarta EE 10 and Security 3.0"
+author: andrew-hughes
+by: contractor
+communities: [java,security]
+description: "Use Jakarta EE 10 to build a secure Java web application using Open ID connect and Auth0."
+tags: [java, jakartaee]
+tweets:
+- ""
+- ""
+- ""
+image:
+type: conversion
+---
+
+## Open ID authentication with Jakarta EE 10 and Security 3.0
+
+Jakarta EE 10 includes a new authentication mechanism: Open ID Connect! This can be added to a Jakarta EE servlet using the new `@OpenIdAuthenticationMechanismDefinition` annotation. 
+
+In this tutorial, you are going to see how to implement a web application with Open ID Connect (OIDC) authentication and use Auth0 as the OIDC provider. You are also going to see one way to secure an API and authenticate using JSON Web Tokens (JWTs). This will all be accomplished using Wildfly as the Jakarta EE runtime.
+
+This stack includes a lot of technologies. I'm going to introduce them briefly below. If you're comfortable with all those terms and just want to get to the code, **feel free to skip ahead to the requirements section**. 
+
+**Jakarta vs Java, EE vs SE**
+
+Jakarta EE is Jakarta Enterprise Edition. This was formerly Java EE. The name and framework packages were migrated when Oracle gave Java EE to the Eclipse Foundation because Oracle still has the rights to the Java brand and did not open-source absolutely everything that was in the `javax.*` namespace. Thus, Jakarta EE is the Eclipse-owned and now totally open-source Java EE (You may have recently had to change some packages from `javax` to `jakarta`. This is why.)
```

Comment:
```suggestion
Jakarta EE is Jakarta Enterprise Edition, formerly known as Java EE. The name and framework packages were migrated when Oracle gave Java EE to the Eclipse Foundation because Oracle still has the rights to the Java brand and did not open-source absolutely everything that was in the `javax.*` namespace. Thus, Jakarta EE is the Eclipse-owned and now totally open-source Java EE (You may have recently had to change some packages from `javax` to `jakarta`. This is why.)
```